### ISSUE TITLE
Ccurrently the "down arrow" work as the "right arrrow" opening the sections when you overpass them. 

### ISSUE THREAD

**Issue by gprieto** [aquirdTurtle/Collapsible_Headings#10]
The bugs mentioned by @Tabixe are still active. I imagined that because he commented in an already closed issue, these have not been noticed.

Therefore I open a new issue, as these needs fixing ASAP. I believe it is just a matter of changing shortcuts behaviour. 

**1.  Issue with the "down arrow":** currently the "down arrow" work as the "right arrrow" opening the sections when you overpass them. 
https://drive.google.com/file/d/1enm2zd83Frkg7V72vRnNPzhiqIrF4uKm/view?usp=sharing

**2. Issue with the "up arrow":** as the "down arrow" it also opens the sections when you overpass them.
https://drive.google.com/file/d/1WKNJ6ztjz6W_tf1Y9Ru3MIWeIBnozW25/view?usp=sharing

**3. Issue with the subsections:** when you are in a header of a subsection you are not able to go one level up by pressing the '"left arrow"
https://drive.google.com/file/d/1fVQ7YKtl-HoaJyZhDg9o8Nea1Zn27y4o/view?usp=sharing

**4. Issue with the relocation of the scroll bar:** when you close a really big section, the scroll bar doesn't get adjusted automatically making impossible see where are you.
https://drive.google.com/file/d/1NLMyBbn2ZX6APAxpDgNVHjbyIUsETdwg/view?usp=sharing

_Originally posted by @Tabixe in https://github.com/aquirdTurtle/Collapsible_Headings/issues/8#issuecomment-545499560_

**Comment by aquirdTurtle** [comment]
> The bugs mentioned by @Tabixe are still active. I imagined that because he commented in an already closed issue, these have not been noticed.

I had actually seen them and just hadn't had time to work on this repo, but thanks for opening the dedicated issue for them anyways.

> Therefore I open a new issue, as these needs fixing ASAP. 

If these things are important to you, you're always welcome to contribute yourself. It's not like I get paid in any way to maintain this repo. I work on it when I have time and am in the mood.

Anyways, I believe I have fixed issues 1, 2, and 3 in version 0.5.0, available now.  Please let me know if they aren't working for you or if there are any additional issues. I'm a little nervous about adding behavior to up-arrow and down-arrow keys since jupyterlab already handles them, but my modifications seem to be working smoothly for me. 

For issue 4, I'm waiting on https://github.com/jupyterlab/jupyterlab/pull/6818 to be merged before I fix this, as a "scroll to cell" method is exactly what I want here. I don't feel like it's worth trying to use the existing scrolling utilities when the perfect utility for the job is coming down the pipeline. Until then of course you can always get jupyterlab to scroll the cell into view by hitting up-arrow then down-arrow to re-select the cell after closing it.

I'll leave the issue open until I deal with issue 4.

**Comment by Tabixe** [comment]
Thanks a lot @aquirdTurtle! I hope I knew more coding so I could help you. I have tried the up-to-date version and works perfectly!

You are a master!

**Comment by gprieto** [comment]
Indeed all works as intended, thank you @aquirdTurtle! Nice job!

**Comment by Tabixe** [comment]
Hi @aquirdTurtle,

I'm writing here because is the only way I have to get in touch with you. You have done an amazing job with the Collapsible Headings but I'm going to push my luck and suggest you another challenge for the next time that you fill in the mood of coding. 

I don't know at all how difficult it is but it would be incredible if you manage to create a new Jupyter Lab extension that provides the run time for each cell and when they were ran for last time.

![image](https://user-images.githubusercontent.com/43148181/68584117-a0e16700-0476-11ea-8229-a5488f63f14d.png)

If you managed to do this I think that we would be at the point where we can start saying that jupyter lab is better than jupyter notebook. I only know python and SQL but if I can help you with something I will do it for sure.

Have a great week and Thanks a lot!

**Comment by aquirdTurtle** [comment]
@Tabixe 

Hi there, I've not made any progress on such an extension, although I might consider it at some point, but I thought I'd point you to the fact that the data is at least obtainable by looking at the cell metadata as of Jupyterlab version 1.2:
https://github.com/jupyterlab/jupyterlab/pull/6864
but there's no UI development as of right now as far as I can tell.

**Comment by aquirdTurtle** [comment]
I got a bit tired of waiting for the main Jupyterlab repo to add this feature, so I went head and fixed this. As of version 2.2.0, just published, after closing or opening a large header, the notebook will scroll to keep the header in view if needed. Let me know if there are any additional issues.

**Comment by simonkeys** [comment]
Issues 1 and 2 from OP are still issues for me. Is that a regression?

I'm using aquirdturtle-collapsible-headings-3.0.1, jupyterlab 3.0.0

**Comment by aquirdTurtle** [comment]
@simonkeys it shouldn't be, but there has been a little bug where occasionally if jupyterlab takes an especially long time to load for some reason, the shortcut doesn't register, but should generally work if you simply reload the page. Is it possible that this is your issue as well? I did notice this getting a bit worse with jupyterlab 3, and should probably just fix it regardless.

**Comment by simonkeys** [comment]
Yes, reloading the page does the trick, thanks! 

I've got JupyterLab configured to run in Chrome's "application mode" (like this: http://christopherroach.com/articles/jupyterlab-desktop-app/). Maybe that triggers the not-registering problem somehow?

**Comment by aquirdTurtle** [comment]
I just tried out the application mode version, and it does seem like it's maybe a bit slower to load than otherwise, so might be exacerbating the problem. I'll reopen this and try to fix this properly.

**Comment by simonkeys** [comment]
It looks like the slow loading may actually have been due to another issue -- I had warnings on load because of this: https://github.com/ipython-contrib/jupyter_contrib_nbextensions/issues/1529

After fixing that, the arrow shortcuts now work properly for me on the first load.

Thanks for your help, and for the super useful extension!

**Comment by gsaurabhr** [comment]
I started seeing this issue after updating to jupyterlab 3. Is this common, or just something going wrong with the upgrade?
